tests/php/ORM/DataObjectSchemaTest.php

```diff
@@ -343,4 +343,35 @@ public function testFieldsCanBeIndexedFromFieldSpecs()
             'columns' => ['IndexedMoneyCurrency', 'IndexedMoneyAmount']
         ], $indexes['IndexedMoney']);
     }
+
+    /**
+     * Ensure that records with unique indexes can be written
+     */
+    public function testWriteUniqueIndexes()
+    {
+        // Create default object
+        $zeroObject = new AllIndexes();
+        $zeroObject->Number = 0;
+        $zeroObject->write();
+
+        $this->assertListEquals(
+            [
+                ['Number' => 0],
+            ],
+            AllIndexes::get()
+        );
+
+        // Test a new record can be created without clashing with default value
+        $validObjcet = new AllIndexes();
+        $validObjcet->Number = 1;
+        $validObjcet->write();
```

Comment:
![](https://www.commitstrip.com/wp-content/uploads/2018/11/Strip-Surtout-ne-touche-%C3%A0-rien-650-finalenglish.jpg)

Eh, I'll fix this in a second.